### ISSUE TITLE
/sbin/my_init exited 111

### ISSUE THREAD

**Issue by Redsandro** [cpoppema/docker-flexget#22]
I've been running this for some time, but since the last reboot, it's not executing and I'm getting the error `/sbin/my_init exited 111` with this error log:

```
exec: fatal: unable to exec /sbin/my_init: No such file or directory
/usr/lib/python2.7/site-packages/requests/__init__.py:80: RequestsDependencyWarning: urllib3 (1.22) or chardet (2.3.0) doesn't match a supported version!
  RequestsDependencyWarning)
usermod: no changes
exec: fatal: unable to exec /sbin/my_init: No such file or directory
```

Here is the tail of the normal log:

```
Requirement already up-to-date: pycparser in /usr/lib/python2.7/site-packages (from cffi>=1.7->cryptography>=2.1.4->pyopenssl>=0.12->service_identity)
Installing collected packages: six, idna
  Found existing installation: six 1.10.0
    Uninstalling six-1.10.0:
      Successfully uninstalled six-1.10.0
  Found existing installation: idna 2.5
    Uninstalling idna-2.5:
      Successfully uninstalled idna-2.5
Successfully installed idna-2.6 six-1.11.0
[cont-init.d] 22-update-deluge: exited 0.
[cont-init.d] 30-set-web-passwd: executing... 
[cont-init.d] 30-set-web-passwd: exited 0.
[cont-init.d] done.
[services.d] starting services
[services.d] done.
/sbin/my_init exited 111
[cont-finish.d] executing container finish scripts...
[cont-finish.d] done.
[s6-finish] syncing disks.
[s6-finish] sending all processes the TERM signal.
[s6-finish] sending all processes the KILL signal and exiting.
```

When I recreate the container, it takes longer to start because it installs everything, but it exits with the same error message.

Any idea why my installation broke?

**Comment by cpoppema** [comment]
Hi @Redsandro. Sounds very odd indeed!

Do you have a custom script file to issue any commands after you've restarted/recreated the container ?  What are the commands you ran to reboot/recreate/start the container? 

Since the switch from phusion to s6 overlay, /sbin/my_init no longer exists, since it's a different system.

```
$ docker pull cpoppema/docker-flexget
Using default tag: latest
latest: Pulling from cpoppema/docker-flexget
Digest: sha256:fb858d2c2daab742626a99954bf9d5c10f320c2791cb2c95847091be104bb59e
Status: Image is up to date for cpoppema/docker-flexget:latest
```

Do you also see this error before/after you've pulled the latest and run this command ?
```
$ docker run --rm -v /tmp/config:/config cpoppema/docker-flexget echo Hello World
[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] 10-adduser: executing...
usermod: no changes

[...]

[cont-init.d] 10-adduser: exited 0.
[cont-init.d] 20-update-flexget: executing...

[...]

Successfully installed aniso8601-1.2.1 apscheduler-3.3.1 babelfish-0.5.5 beautifulsoup4-4.6.0 certifi-2017.4.17 chardet-3.0.3 cheroot-5.5.0 cherrypy-10.2.2 click-6.7 colorclass-2.2.0 feedparser-5.2.1 flask-0.12.2 flask-compress-1.4.0 flask-cors-3.0.2 flask-login-0.4.0 flask-restful-0.3.6 flask-restplus-0.10.1 flexget-2.11.15 funcsigs-1.0.2 functools32-3.2.3.post2 future-0.16.0 futures-3.2.0 guessit-2.0.4 html5lib-0.999999999 idna-2.5 itsdangerous-0.24 jinja2-2.9.6 jsonschema-2.6.0 markupsafe-1.0 path.py-10.3.1 pathlib-1.0.1 plumbum-1.6.3 portend-1.8 pynzb-0.1.0 pyparsing-2.2.0 pyrss2gen-1.1 python-dateutil-2.6.1 pytz-2017.2 pyyaml-3.12 rebulk-0.8.2 rpyc-3.3.0 six-1.10.0 sqlalchemy-1.1.10 tempora-1.8 terminaltables-3.1.0 tzlocal-1.4 webencodings-0.5.1 werkzeug-0.12.2 zxcvbn-python-4.4.15
[cont-init.d] 20-update-flexget: exited 0.
[cont-init.d] 21-update-transmission: executing...

[...]

Successfully installed six-1.11.0 transmissionrpc-0.11
[cont-init.d] 21-update-transmission: exited 0.
[cont-init.d] 22-update-deluge: executing...
[cont-init.d] 22-update-deluge: exited 0.
[cont-init.d] 30-set-web-passwd: executing...
[cont-init.d] 30-set-web-passwd: exited 0.
[cont-init.d] done.
[services.d] starting services
[services.d] done.
Hello World
echo exited 0
[cont-finish.d] executing container finish scripts...
[cont-finish.d] done.
[s6-finish] syncing disks.
[s6-finish] sending all processes the TERM signal.
[s6-finish] sending all processes the KILL signal and exiting.
```

**Comment by Redsandro** [comment]
Hi @cpoppema thank you for taking the time to respond.

I was collecting the data you requested and found no difference in our outputs (except for extra debug messages that my configuration is utf-8). And to my surprise, when I rebuilt the container that I tried some times before using the command I always use, I found that it's working this time.

Since I tried a couple of times, I am not sure what changed, or what I did wrong. It is as if the new image was executing with some residual old provision files. But I am not sure why the problem went away, because I have not actively done anything different, except for executing the Hello World line from your post above.

I'm happy it's running now, but I am still puzzled as to what caused and solve this. With that, I think I should close this issue and re-open this if it happens again.

Thanks again!
